### PR TITLE
Traversal must clear scoped roles when backtracking closures

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?

A nondeterministic traversal execution bug was discovered by enabling ignored tests in https://github.com/graknlabs/behaviour/pull/183. This PR replicates the failure as a Traversal integration test and correctly clears scopes when a closure succeeds, but the next branch fails.

## What are the changes implemented in this PR?
* pop scopes when a closure succeeds but then the next recursive branch/closure fails, in `computeFirst`
* replicate and keep the traversal test for this case and add to CI
* bump behaviour repository to the latest master

### CODE REVIEW

File: test/integration/BasicTest.java [hunk 1]
```diff
@@ -818,12 +818,12 @@ public void test_query_cancelled_asynchronously() throws IOException {
...
                         session.close();
                     }).start();
-                    tx.query().match(Graql.parseQuery("match $x isa thing;").asMatch());
+                    tx.query().match(Graql.parseQuery("match $x sub thing;").asMatch());
                 }).start();
             }
             Grakn.Session session = grakn.session(database, Arguments.Session.Type.DATA);
             Grakn.Transaction tx = session.transaction(Arguments.Transaction.Type.WRITE);
-            tx.query().match(Graql.parseQuery("match $x isa thing;").asMatch());
+            tx.query().match(Graql.parseQuery("match $x sub thing;").asMatch());
```
Comment (haikalpribadi) [inline test/integration/BasicTest.java:826]: Why do we need to change this, @flyingsilverfin ?

**Review by haikalpribadi — COMMENTED** [review]
All looks good, just got one comment below

File: test/integration/BasicTest.java (hunk 1 above)
Comment (flyingsilverfin) [reply test/integration/BasicTest.java:826]: errr I can move this to my other PR, however in the other PR type resolver complains this query is unsatisfiable - which is correct, there are no concrete subtypes of `thing`.

In the other PR i've made the "no abstract types are allowed on Thing variables" smarter than before, which is why it picks this up now. However probably makes sense to move this change over there!

Comment (flyingsilverfin) [reply test/integration/BasicTest.java]: removed from this PR

Comment (haikalpribadi) [reply test/integration/BasicTest.java]: yes it does make sense to keep that change scoped in the right PR, @flyingsilverfin

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**